### PR TITLE
[feat](nereids) add session variable disable_nereids_expression_rules

### PR DESCRIPTION
**Pull request by yujun777**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

Add session variable test_disable_nereids_expression_rule for disable  expression rule, example:

```sql
SET disable_nereids_expression_rules='SIMPLIFY_COMPARISON_PREDICATE'
```

for a expression rule `ExpressionPatternMatcher`,  its has default rule type:  ANONYMOUS.

when it calls `toRule(ruleTypeXX)` will create a rule with type:  ruleTypeXX.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by englefly** [comment]
some rules are directly invoked by other rules. For such case, this pr cannot disable them.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/ExpressionRuleType.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.expression;
+
+/**
+ * Type of rewrite expression rules.
+ */
+public enum ExpressionRuleType {
+    SIMPLIFY_COMPARISON_PREDICATE,
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/ExpressionRuleType.java:24]: only this rule can be disabled?
how about other expression rewrite rules?

Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/ExpressionRuleType.java:44]: > only this rule can be disabled? how about other expression rewrite rules?

add rule type

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run feut

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run p0

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run performance

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run feut

**Review by 924060929 — DISMISSED** [review]

**Comment by yujun777** [comment]
run buildall

**Review by 924060929 — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**